packages/x-tree-view-pro/src/internals/plugins/useTreeViewItemsReordering/useTreeViewItemsReordering.test.tsx

```diff
@@ -46,11 +46,14 @@ const buildTreeViewDragInteractions = (dataTransfer: DataTransfer) => {
     fullDragSequence: (
       draggedItem: HTMLElement,
       targetItem: HTMLElement,
-      options: DragEventOptions = {},
+      options: DragEventOptions & { beforeDragEnd?: (dataTransfer: DataTransfer) => void } = {},
```

Comment:
Is it possible to move this new property inside `DragEventOptions`?
And add a small JSDoc (even if the name is quite explicit)